File: src/main/java/com/melonbase/microquark/service/HelloService.java
package com.melonbase.microquark.service;

import com.melonbase.microquark.repo.HelloRepo;

import javax.enterprise.context.ApplicationScoped;
import javax.inject.Inject;

@ApplicationScoped
public class HelloService {

  @Inject
  HelloRepo helloRepo;

  public String getHello() {
    return helloRepo.getHello();
  }

  public void setHello(final String msg) {
    helloRepo.setHello(msg);
  }
}

File: src/main/code/com/melonbase/microquark/service/HelloService.kt
package com.melonbase.microquark.service

import com.melonbase.microquark.repo.HelloRepo
import javax.enterprise.context.ApplicationScoped
import javax.inject.Inject

@ApplicationScoped
class HelloService @Inject constructor(private val helloRepo: HelloRepo) {

  fun getHello(): String? {
    return helloRepo.getHello()
  }

  fun setHello(msg: String) {
    helloRepo.setHello(msg)
  }
}